src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -1178,5 +1178,195 @@ private class MyClassCustomConverterFactory : JsonConverterFactory
                 return ConverterInstance;
             }
         }
+
+        [Fact]
+        public static void ClassWithExtensionDataAttribute_IsReturnedInMetadata()
+        {
+            var resolver = new DefaultJsonTypeInfoResolver();
+            var jti = resolver.GetTypeInfo(typeof(ClassWithExtensionDataAttribute), new());
+
+            Assert.Equal(3, jti.Properties.Count);
+            Assert.False(jti.Properties[0].IsExtensionData);
+            Assert.False(jti.Properties[1].IsExtensionData);
+
+            JsonPropertyInfo lastProperty = jti.Properties[2];
+            Assert.Equal("ExtensionData", lastProperty.Name);
+            Assert.Equal(typeof(Dictionary<string, object>), lastProperty.PropertyType);
+            Assert.True(lastProperty.IsExtensionData);
+        }
+
+        [Fact]
+        public static void ClassWithExtensionDataAttribute_ChangingExtensionDataFlagDisablesExtensionData()
+        {
+            var options = new JsonSerializerOptions
+            {
+                TypeInfoResolver = new DefaultJsonTypeInfoResolver()
+                {
+                    Modifiers =
+                    {
+                        jti =>
+                        {
+                            if (jti.Type == typeof(ClassWithExtensionDataAttribute))
+                            {
+                                Assert.Equal(3, jti.Properties.Count);
+                                Assert.True(jti.Properties[2].IsExtensionData);
+                                jti.Properties[2].IsExtensionData = false;
```

Comment:
do we need reverse test? changing one extension data into the other property? Also I'm missing functional test where user incorrectly set two extension data properties and we fix that

Reply:
See https://github.com/dotnet/runtime/pull/71797#discussion_r916751712. Having two attributes will currently fail early. I will add a test that causes `Configure` to fail if the resolver specifies duplicate properties with `IsExtensionData`.